src/analysis/classic/warlock/demonology/modules/spells/Doomguard.tsx

```diff
@@ -0,0 +1,394 @@
+      this.doomguardId = doomguardId;
+    }
+
+    this.snapshotQualityEntries.forEach((entry) => {
+      console.log(entry.issueWithItemOrEnchant);
+      if (!entry.issueWithItemOrEnchant) {
+        entry.relatedBuffs?.forEach((relatedBuff) => {
+          if (this.selectedCombatant.hasBuff(relatedBuff.id, event.timestamp)) {
+            entry.snapshotQuality = QualitativePerformance.Good;
+            entry.snapshotSummary = (
+              <li>
+                <PerformanceMark perf={entry.snapshotQuality} />{' '}
+                <SpellLink spell={relatedBuff.id} />{' '}
+                {entry.item ? (
+                  <>
+                    {' '}
+                    from{' '}
+                    <ItemLink id={entry.item.id} details={entry.item} icon={false}>
+                      <Icon icon={entry.item.icon} />
+                    </ItemLink>
+                  </>
+                ) : (
+                  <></>
+                )}
+              </li>
+            );
+          }
+          console.log(entry);
+          if (entry.snapshotQuality === QualitativePerformance.Fail) {
+            entry.snapshotSummary = (
+              <li>
+                <PerformanceMark perf={entry.snapshotQuality} />{' '}
+                <SpellLink spell={entry.relatedBuffs[0].id} />{' '}
+                {entry.item ? (
+                  <>
+                    {' '}
+                    from{' '}
+                    <ItemLink id={entry.item.id} details={entry.item} icon={false}>
+                      <Icon icon={entry.item.icon} />
+                    </ItemLink>
+                  </>
+                ) : (
+                  <></>
+                )}
+              </li>
+            );
+          }
+        });
+      }
+    });
+
+    const failedEntries = this.snapshotQualityEntries.filter(
+      (entry) => entry.snapshotQuality === QualitativePerformance.Fail,
+    );
+
+    console.log('failedEntries', failedEntries);
+
+    switch (failedEntries.length) {
+      case 0:
+        this.snapshotAdvice = <>You snapshotted every possible buff! Perfect!</>;
+        break;
+      case 1:
+        console.log(failedEntries[0].relatedBuffs[0]);
+        console.log(enchanting.HURRICANE_BUFF);
+        if (failedEntries[0].relatedBuffs[0] === enchanting.HURRICANE_BUFF) {
+          this.snapshotAdvice = (
+            <>
+              You snapshotted every possible buff, unless{' '}
+              <SpellLink spell={enchanting.HURRICANE_BUFF} />. <br />
+              You can proc both <SpellLink spell={enchanting.POWER_TORRENT_BUFF} /> and{' '}
+              <SpellLink spell={enchanting.HURRICANE_BUFF} /> by casting{' '}
+              <SpellLink spell={SPELLS.SOUL_HARVEST} /> prepull, with a weapon enchanted with
+              Hurricane, then swap to your normal weapon before entering combat. It has only 15%
+              chance to happen, but it adds ~3.5% haste to your doomguard if you succeed!
+            </>
```

Comment:
grammar / formatting
```suggestion
            <>
              <p>You snapshotted every possible buff except{' '}
              <SpellLink spell={enchanting.HURRICANE_BUFF} />.</p>
              <p>You can proc both <SpellLink spell={enchanting.POWER_TORRENT_BUFF} /> and{' '}
              <SpellLink spell={enchanting.HURRICANE_BUFF} /> by:</p>
              <ol>
              <li>Casting <SpellLink spell={SPELLS.SOUL_HARVEST} /> pre-pull with a weapon enchanted with
              Hurricane
              <li>Swapping to your normal weapon before entering combat.
              </ol>
              <p>This has a ~15% chance to give your Doomguard an additional ~3.5% haste.</p>
            </>
```